server/server.js

```diff
@@ -11,6 +13,101 @@ const db = new pg.Pool({
 });
 
 const app = express();
+app.use(express.json());
+app.use(errorMiddleware);
+
+const queryObject = {
+  get: `
+  select *
+    from "entries"
+    order by "entryId";
+  `,
+
+  post: `
+  insert into "entries" ("title", "notes", "photoUrl")
+    values ($1, $2, $3)
+  returning *;
+  `,
+
+  put: `
+  update "entries"
+  set "title"=$1,
+      "notes"=$2,
+      "photoUrl"=$3
+  where "entryId"=$4
+  returning *;
+  `,
+
+  delete: `
+  delete
+  from "entries"
+  where "entryId"=$1
+  returning *;
+  `,
+};
+
+// Routing functions
+app.get('/api/entries', async (req, res, next) => {
+  try {
+    const sql = queryObject.get;
+    const result = await db.query(sql);
+    res.status(200).json(result.rows);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.post('/api/entries', async (req, res, next) => {
+  try {
+    const sql = queryObject.post;
+    const { title, notes, photoUrl } = req.body;
+    const params = [title, notes, photoUrl];
+    const result = await db.query(sql, params);
+    const [entry] = result.rows;
+    res.status(201).json(entry);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.put('/api/entries/:entryId', async (req, res, next) => {
+  try {
+    const entryId = Number(req.params.entryId);
+    if (!Number.isInteger(entryId) || entryId <= 0) {
+      throw new ClientError(400, 'entryId must be a positive integer');
+    }
+    const { title, notes, photoUrl } = req.body;
+    const sql = queryObject.put;
+    const params = [title, notes, photoUrl, entryId];
+    const result = await db.query(sql, params);
+    const [entry] = result.rows;
+    if (!entry) {
+      throw new ClientError(404, `Unable to find entryId ${entryId}`);
+    }
+    res.status(201).json(entry);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.delete('/api/entries/:entryId', async (req, res, next) => {
+  try {
+    const entryId = Number(req.params.entryId);
+    if (!Number.isInteger(entryId) || entryId <= 0) {
+      throw new ClientError(400, 'entryId must be a positive integer');
+    }
+    const sql = queryObject.delete;
+    const params = [entryId];
+    const result = await db.query(sql, params);
+    const [entry] = result.rows;
+    if (!entry) {
+      throw new ClientError(404, `Unable to find entryId ${entryId}`);
+    }
+    res.status(201).json(entry);
```

Comment:
201 means created. For delete, we usually send 204 with no body. But if you want to send the deleted entry back, use 200.